typescript/TypeScriptParser.g4

```diff
@@ -0,0 +1,834 @@
+    | While '(' expressionSequence ')' statement                                                                # WhileStatement
+    | For '(' expressionSequence? SemiColon expressionSequence? SemiColon expressionSequence? ')' statement     # ForStatement
+    | For '(' varModifier variableDeclarationList SemiColon expressionSequence? SemiColon expressionSequence? ')'
+          statement                                                                                             # ForVarStatement
+    | For '(' singleExpression (In | Identifier{this.p("of")}?) expressionSequence ')' statement                # ForInStatement
+    | For '(' varModifier variableDeclaration (In | Identifier{this.p("of")}?) expressionSequence ')' statement # ForVarInStatement
+    ;
+
+varModifier
+    : Var
+    | Let
+    | Const
+    ;
+
+continueStatement
+    : Continue ({this.notLineTerminator()}? Identifier)? eos
+    ;
+
+breakStatement
+    : Break ({this.notLineTerminator()}? Identifier)? eos
+    ;
+
+returnStatement
+    : Return ({this.notLineTerminator()}? expressionSequence)? eos
+    ;
+
+yieldStatement
+    : Yield ({this.notLineTerminator()}? expressionSequence)? eos
+    ;
+
+withStatement
+    : With '(' expressionSequence ')' statement
+    ;
+
+switchStatement
+    : Switch '(' expressionSequence ')' caseBlock
+    ;
+
+caseBlock
+    : '{' caseClauses? (defaultClause caseClauses?)? '}'
+    ;
+
+caseClauses
+    : caseClause+
+    ;
+
+caseClause
+    : Case expressionSequence ':' statementList?
+    ;
+
+defaultClause
+    : Default ':' statementList?
+    ;
+
+labelledStatement
+    : Identifier ':' statement
+    ;
+
+throwStatement
+    : Throw {this.notLineTerminator()}? expressionSequence eos
+    ;
+
+tryStatement
+    : Try block (catchProduction finallyProduction? | finallyProduction)
+    ;
+
+catchProduction
+    : Catch '(' Identifier ')' block
+    ;
+
+finallyProduction
+    : Finally block
+    ;
+
+debuggerStatement
+    : Debugger eos
+    ;
+
+functionDeclaration
+    : Function Identifier callSignature ( ('{' functionBody '}') | SemiColon)
+    ;
+
+//Ovveride ECMA
+classDeclaration
+    : Abstract? Class Identifier typeParameters? classHeritage classTail
+    ;
+
+classHeritage
+    : classExtendsClause? implementsClause?
+    ;
+
+classTail
+    :  '{' classElement* '}'
+    ;
+
+classExtendsClause
+    : Extends typeReference
+    ;
+
+implementsClause
+    : Implements classOrInterfaceTypeList
+    ;
+
+// Classes modified
+classElement
+    : constructorDeclaration
+    | propertyMemberDeclaration
+    | indexMemberDeclaration
+    | statement
+    ;
+
+propertyMemberDeclaration
+    : propertyMemberBase propertyName typeAnnotation? initializer? SemiColon
+    | propertyMemberBase propertyName callSignature ( ('{' functionBody '}') | SemiColon)
+    | propertyMemberBase (getAccessor | setAccessor)
+    | abstractDeclaration
+    ;
```

Comment:
```antlr
propertyMemberDeclaration
    : propertyMemberBase (propertyName ( typeAnnotation? initializer? SemiColon
                                       | callSignature ( ('{' functionBody '}') | SemiColon))
                         | getAccessor
                         | setAccessor)
    | abstractDeclaration
    ;
```

Reply:
I'm not sure that is simplify grammer...

Reply:
Ok, it's not necessary.